packages/babel-plugin-transform-remove-unused-catch-binding/src/index.js

```diff
@@ -0,0 +1,28 @@
+import syntaxOptionalCatchBinding from "babel-plugin-syntax-optional-catch-binding";
+
+export default function() {
+  return {
+    inherits: syntaxOptionalCatchBinding,
+
+    visitor: {
+      CatchClause(path) {
+        const binding = path.scope.getOwnBinding(path.node.param.name);
+        if (
+          binding.constantViolations.filter(el => {
+            return el.node.left.name === path.node.param.name;
+          }).length > 0
+        ) {
+          return;
+        }
+        if (
+          path.node.param &&
+          !binding.referenced &&
+          path.scope.hasBinding(path.node.param.name)
```

Comment:
I think this is not required here. `binding` is already defined, and `path.scope.hasBinding` is going to return `true` always. Did you face any issues without this check?